### PR TITLE
feat(fastestvpn): support for Wireguard

### PR DESCRIPTION
**Pull request by qdm12**
Credits to:

- @Zerauskire for the initial investigation
- @jvanderzande for an initial implementation as well as reviewing this pull request

### CODE REVIEW

**Comment by jvanderzande** [comment]
I've tried to add this commit to my local master and build the local docker image to test, but get an error with these env:
```
      - VPN_SERVICE_PROVIDER=fastestvpn
      - VPN_TYPE=wireguard
      - WIREGUARD_ENDPOINT_PORT=51820
      - SERVER_COUNTRIES=Belgium
      - WIREGUARD_PRIVATE_KEY=-----mylprivatekey-----2=
      - WIREGUARD_ADDRESSES=172.16.125.72/32

```

```
2024-07-30T21:56:21Z INFO [routing] default route found: interface eth0, gateway 172.30.0.1, assigned IP 172.30.0.2 and family v4
2024-07-30T21:56:21Z INFO [routing] local ethernet link found: eth0
2024-07-30T21:56:21Z INFO [routing] local ipnet found: 172.30.0.0/16
2024-07-30T21:56:21Z INFO [firewall] enabling...
2024-07-30T21:56:21Z INFO [firewall] enabled successfully
2024-07-30T21:56:22Z INFO [storage] merging by most recent 19738 hardcoded servers and 19738 servers read from /gluetun/servers.json
2024-07-30T21:56:22Z ERROR VPN settings: provider settings: server selection: Wireguard server selection settings: endpoint port is set
2024-07-30T21:56:22Z INFO Shutdown successful
```

Any idea what's wrong?

**Comment by jvanderzande** [comment]
Tried this morning to just checkout just the "fastestvpn-wireguard" branch and build the docker image and this version does start fine with the above env.  Not sure why the other test failed, but this version seems to work fine for FastestVpn wireguard!

Thanks! Jos :)

**Comment by jvanderzande** [comment]
Just had some time to do a bit more testing and found out I used the wrong branch for testing this morning. This time used your "fastestvpn-wireguard)" and the rebuild image gives the exact same error as shown in the first response.

**Comment by jvanderzande** [comment]
Things seem to work fine now after I changed the below line in **connection.go**, changing the 3rd param from 0 to 51820:
`	defaults := utils.NewConnectionDefaults(4443, 4443, 51820) //nolint:gomnd
`

**Comment by qdm12** [comment]
Thanks @jvanderzande for testing and even finding the problem! Although, I'm a bit confused when you say in your first comment

> I've tried to add this commit to my local master and build the local docker image to test, but get an error with these env:
> - WIREGUARD_ENDPOINT_PORT=51820

And then in your 2nd comment

> It now also works without defining the env: WIREGUARD_ENDPOINT_PORT=51820

So before the 'fix', just to be sure, it would work fine if you would set `WIREGUARD_ENDPOINT_PORT=51820` correct? I'm asking since I did a recent change regarding this in 36c8da7ea7152d71ba62c5f1884784793d4facde so just double checking it works as expected. Thanks!

**Comment by jvanderzande** [comment]
> Thanks @jvanderzande for testing and even finding the problem! Although, I'm a bit confused when you say in your first comment
> 
> > I've tried to add this commit to my local master and build the local docker image to test, but get an error with these env:
> > 
> > * WIREGUARD_ENDPOINT_PORT=51820
> 
> And then in your 2nd comment
> 
> > It now also works without defining the env: WIREGUARD_ENDPOINT_PORT=51820
> 

OK...  will give you the whole story of my testing:
In my first test without the WIREGUARD_ENDPOINT_PORT definition, I got an error Hard Error and some message that the Wireguard port was missing, so added the - WIREGUARD_ENDPOINT_PORT=51820 environment setting to the docker compose, which then triggered the error I've shared. (I assumed at the time that WIREGUARD_ENDPOINT_PORT was a required field)
Then I figured out that the default port was missing as indicated, which made all work fine and now also when I do not provide the WIREGUARD_ENDPOINT_PORT environment variable.   
Hope I described it a bit clearer, but shoot when you still have questions.

**Comment by jvanderzande** [comment]
Just tested the latest version of this branch in my docker test setup and FASTESTVPN works fine for me.

**Comment by qdm12** [comment]
Oh ok never mind, I didn't read correctly the error you shared `endpoint port is set` which is meant to happen since custom ports should not be allowed (AFAIK) for FastestVPN wireguard. All good 💯 And yeah because gluetun didn't know what port to use, it "panic"-ed with an error crashing the whole thing. Merging it! 👍

File: internal/portforward/service/start.go [hunk 1]
```diff
@@ -79,9 +79,12 @@ func (s *Service) Start(ctx context.Context) (runError <-chan error, err error)
 	keepPortDoneCh := make(chan struct{})
 	s.keepPortDoneCh = keepPortDoneCh
 
+	readyCh := make(chan struct{})
```
Comment (qdm12) [inline internal/portforward/service/start.go:82]: revert

File: internal/provider/fastestvpn/updater/servers.go [hunk 2]
```diff
@@ -5,33 +5,38 @@ import (
...
 		}
 	}
 
-	if len(hts) < minServers {
-		return nil, fmt.Errorf("%w: %d and expected at least %d",
-			common.ErrNotEnoughServers, len(hts), minServers)
-	}
-
 	hosts := hts.toHostsSlice()
+
```
Comment (qdm12) [inline internal/provider/fastestvpn/updater/servers.go:39]: remove empty line

**Comment by bijfy** [comment]
It's working great with wireguard fastestvpn.  Thank you so much everyone for your great work. I am.getting 560+ mbps on mobile and around 780+ mbps on my wired 1gig plan on zimaboard.

**Pull request merged**